### ISSUE TITLE
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: content_contentnode_tags.id

### ISSUE THREAD

**Issue by lyw07** [learningequality/kolibri#2314]
## Summary

When I'm trying to import a channel after another one, the importing failed with the error listed in `Traceback` section

## System information

*Please specify the Kolibri version you were using and your operating system!*

 - Version: kolibri-0.6.dev020170921163803-git.pex from https://storage.googleapis.com/le-downloads/kolibri%2Frelease%2F2632%2Freport.html
 - Operating system: Mac OS
 - Browser: Google Chrome

I also tested on Ubuntu 14.04 with Firefox and got the same issue

## Traceback or relevant snippet from server.log or browser console
```
INFO     Importing ContentTag data
INFO     Importing ContentNode_has_prerequisite data
INFO     Importing ContentNode_related data
INFO     Importing ContentNode_tags data
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Users/lingyiwang/Documents/Learning Equality/kolibri/kolibri/__main__.py", line 8, in <module>
    main(args=sys.argv[1:])
  File "kolibri/utils/cli.py", line 580, in main
    manage(command, args=django_args)
  File "kolibri/utils/cli.py", line 411, in manage
    execute_from_command_line(argv=argv)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 54, in execute
    super(Command, self).execute(*args, **options)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "kolibri/core/webpack/management/commands/devserver.py", line 61, in handle
    update_channel_metadata()
  File "kolibri/content/utils/annotation.py", line 31, in update_channel_metadata
    import_channel_from_local_db(channel_id)
  File "kolibri/content/utils/channel_import.py", line 305, in import_channel_from_local_db
    import_manager.import_channel_data()
  File "kolibri/content/utils/channel_import.py", line 196, in import_channel_data
    unflushed_rows = self.table_import(model, row_mapper, table_mapper, unflushed_rows)
  File "kolibri/content/utils/channel_import.py", line 184, in table_import
    self.destination.session.bulk_insert_mappings(DestinationRecord, data_to_insert)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2463, in bulk_insert_mappings
    return_defaults, False, render_nulls)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2536, in _bulk_save_mappings
    transaction.rollback(_capture_exception=True)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2531, in _bulk_save_mappings
    isstates, return_defaults, render_nulls)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 69, in _bulk_insert
    bookkeeping=return_defaults)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 799, in _emit_insert_statements
    execute(statement, multiparams)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 945, in execute
    return meth(self, multiparams, params)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 263, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1053, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1189, in _execute_context
    context)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
    exc_info
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1159, in _execute_context
    context)
  File "/Users/lingyiwang/.venvs/kolibri/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 467, in do_executemany
    cursor.executemany(statement, parameters)
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: content_contentnode_tags.id [SQL: u'INSERT INTO content_contentnode_tags (id, contentnode_id, contenttag_id) VALUES (?, ?, ?)'] [parameters: ((1, u'7a8797a4dc4a412b8854c8111a93c0a1', u'0e70b89d2b454c0f83b21148197c8391'), (2, u'7a8797a4dc4a412b8854c8111a93c0a1', u'88bf5d8352a84b0ca3eb9da49f023fbd'), (3, u'7a8797a4dc4a412b8854c8111a93c0a1', u'04f845ad365844a88177141ecbb9e9cc'), (4, u'7a8797a4dc4a412b8854c8111a93c0a1', u'9fac0eefbe2c49b180c6cfbc8ef71ca0'), (5, u'6b3d90cc49e94b9fab7d691099beaa35', u'1eb2b903266e42e092ae0259ba71ff3a'), (6, u'6b3d90cc49e94b9fab7d691099beaa35', u'0e70b89d2b454c0f83b21148197c8391'), (7, u'6b3d90cc49e94b9fab7d691099beaa35', u'88bf5d8352a84b0ca3eb9da49f023fbd'), (8, u'194b533cc6e041028926a718a53eb6bc', u'6e80aec1d2164403a1e2133062e35961')  ... displaying 10 of 86 total bound parameter sets ...  (85, u'9d78f4242202494a82b6ac8d3553c74a', u'968d25449ae24189a5d7cbd6d90f1088'), (86, u'9d78f4242202494a82b6ac8d3553c74a', u'9fac0eefbe2c49b180c6cfbc8ef71ca0'))]
INFO     Closing webpack process
```

## How to reproduce

1. Run kolibri with `python kolibri.pex start`
2. Import channel with id `8b4d3e6d3d4842ba8ea658335b5dd252` from the internet
3. After the first import is done, start to import channel with id `b3556676589f4c408142da1d0f06cebc`
4. After it shows failure of import, do `python kolibri.pex stop`, the error will be shown

**Comment by lyw07** [comment]
@aronasorman @rtibbles

**Comment by rtibbles** [comment]
Note that this is almost certainly due to a previous incomplete import, whereby the ContentNode<->Tags ManyToMany intermediary table has not been properly wiped.

**Comment by indirectlylit** [comment]
possibly fixed?

**Comment by lyw07** [comment]
I can check after the dev meeting

**Comment by lyw07** [comment]
Confirm that it's fixed by the latest 0.7 pex. (Although I tested with command line import as I'm not able to click import button through browser, it should be the same result if testing through the browser).

**Comment by rtibbles** [comment]
It's the particular combination of channels that causes the issue, so should not be a matter of whether it is run via icequebe or not! Thanks for double checking.

**Comment by rtibbles** [comment]
(It was fixed in #2564)
